File: njdot/paths.py
from os.path import join

from nj_crashes.paths import PUBLIC_DIR, ROOT_DIR
from njdot.tbls import parse_type

NJDOT_DIR = join(ROOT_DIR, 'njdot')
DOT_DATA = join(NJDOT_DIR, 'data')
DOT_DATA_S3 = 's3://nj-crashes/njdot/data'


WWW_DOT = join(PUBLIC_DIR, 'njdot')
WWW_DATA_DOT = join(PUBLIC_DIR, 'data', 'njdot')
CNS = f'{WWW_DOT}/with_cns.parquet'
CRASHES_PQT = f'{DOT_DATA}/crashes.parquet'
OCCUPANTS_PQT = f'{DOT_DATA}/occupants.parquet'
PEDESTRIANS_PQT = f'{DOT_DATA}/pedestrians.parquet'
VEHICLES_PQT = f'{DOT_DATA}/vehicles.parquet'
CM_PQT = f'{DOT_DATA}/cm.pqt'
CRASHES_DB = f'{WWW_DOT}/crashes.db'
CC2MC2MN = f'{WWW_DOT}/cc2mc2mn.json'

CMYMC_DB = f'{WWW_DOT}/cmymc.db'

MC_PQT = join(DOT_DATA, 'muni_codes.parquet')

# AASHTO Crash.csv pipeline outputs (`njdot aashto …` subcmds)
AASHTO_COMBINED_CRASHES = f'{DOT_DATA}/aashto_combined_crashes.parquet'
AASHTO_SUPPLEMENTED_CRASHES = f'{DOT_DATA}/aashto_supplemented_crashes.parquet'
AASHTO_SUPPLEMENTED_OCCUPANTS = f'{DOT_DATA}/aashto_supplemented_occupants.parquet'
AASHTO_SUPPLEMENTED_PEDESTRIANS = f'{DOT_DATA}/aashto_supplemented_pedestrians.parquet'
AASHTO_SUPPLEMENTED_VEHICLES = f'{DOT_DATA}/aashto_supplemented_vehicles.parquet'

# NJSP-derived geocode backfill for NJDOT fatals missing both
# `(olat, olon)` and `(sri, mp)`. Sidecar parquet keyed by
# `(year, cc, mc, case)`; merged in by `load_crashes_with_aashto`.
CRASHES_GEOCODE_BACKFILL = f'{DOT_DATA}/crashes_geocode_backfill.parquet'


def aashto_year_path(year: int, name: str) -> str:
    return f'{DOT_DATA}/{year}/{name}'


def raw_pqt_path(tpe, year, county=None):
    county = county or 'NewJersey'
    tpe = parse_type(tpe)
    return f'{DOT_DATA}/{year}/{county}{year}{tpe}.pqt'

File: www/server/paths.ts
import { dirname, join, relative } from "path";
import * as paths from "@/src/paths";
import { DotUrls } from "@/src/urls";

export const WWW = process.cwd()
export const PUBLIC = join(WWW, "public")
export const ROOT = dirname(WWW)
export const NJSP = join(PUBLIC, "njsp")
export const PLOTS = join(PUBLIC, "plots")
export const RUNDATE = join(NJSP, "rundate.json")

export const NJDOT = join(PUBLIC, "njdot")
export const CC2MC2MN = join(NJDOT, "cc2mc2mn.json")
export const CC2MC2MN_RELPATH = relative(WWW, CC2MC2MN)

export const NJSP_DATA = join(ROOT, "njsp", "data")
export const ProjectedCsv = join(PUBLIC, paths.ProjectedCsv)
export const YearTypeCountyCsv = join(NJSP_DATA, "year-type-county.csv")
export const NjspCrashesPqt = join(NJSP_DATA, "crashes.parquet")

export const NJDOT_DATA = join(ROOT, "njdot", "data")
export const DotPqts: DotUrls = {
  crashes: join(NJDOT_DATA, "crashes.parquet"),
  drivers: join(NJDOT_DATA, "drivers.parquet"),
  occupants: join(NJDOT_DATA, "occupants.parquet"),
  pedestrians: join(NJDOT_DATA, "pedestrians.parquet"),
  vehicles: join(NJDOT_DATA, "vehicles.parquet"),
}
